chapter10/mssw/exercise2.pl

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use Data::Dumper;
+use Benchmark qw( :all );
+
+sub human_friendly_sort {
+    my @sorted = sort( { -s $a <=> -s $b } glob('*') );
+}
+
+sub cyborg_friendly_sort {
+    my @sorted =
+      map( $_->[0],
+        sort( { $a->[1] <=> $b->[1] } map( [ $_, -s $_ ], glob('*') ) ) );
+}
```

Comment:
As long as you remember that these operators are right-associative, you get into the mindset of being able to read them right to left.